File: Native/Test/Fabric/CORE-147.js
FC = createFabricClient();
FABRIC = wrapFabricClient(FC);

op = FABRIC.DependencyGraph.createOperator( "op" );
op.setEntryFunctionName('entry');
op.setSourceCode("operator entry( io Scalar input ) { String foo; report foo; }");

binding = FABRIC.DependencyGraph.createBinding();
binding.setOperator( op );
binding.setParameterLayout( [ "self.input" ] );

node = FABRIC.DependencyGraph.createNode( "node" );
node.addMember( "input", "Scalar" );
node.bindings.append( binding );
node.setData( "input", 17 );
node.evaluate();

FC.dispose();

File: Native/Test/Python/CORE-147.py
#
#  Copyright 2010-2012 Fabric Technologies Inc. All rights reserved.
#

import fabric
fabricClient = fabric.createClient()

op = fabricClient.DependencyGraph.createOperator( "op" )
op.setEntryFunctionName('entry')
op.setSourceCode("operator entry( io Scalar input ) { String foo; report foo; }")

binding = fabricClient.DependencyGraph.createBinding()
binding.setOperator( op )
binding.setParameterLayout( [ "self.input" ] )

node = fabricClient.DependencyGraph.createNode( "node" )
node.addMember( "input", "Scalar" )
node.bindings.append( binding )
node.setData( "input", 17 )
node.evaluate()

fabricClient.close()
